### PR TITLE
Add str/repr formatting options and change defaults accordingly

### PR DESCRIPTION
**Pull request by michaelosthege**
+ new options "latex_with_params" and "plain_with_params" replace the current default behavior of including input parameters (going back to default behaviour of the 3.9.3 release)
+ `__latex__` and `_repr_latex` default to "latex_with_params"
+ `__str__` and `_str_repr` default to "plain"
+ `__latex__` repr __of a model__ defaults to "latex" (without params)
+ new formatting kwarg for `model_to_graphviz` enables switching between "plain" (default) and "plain_with_params"

Depending on what your PR does, here are a few things you might want to address in the description:
+ [x] ~~what are the (breaking) changes that this PR makes?~~ - None
+ [x] important background, or details about the implementation
+ [x] are the changes—especially new features—covered by tests and docstrings?
+ [x] ~~consider adding/updating relevant example notebooks~~
+ [x] ~~right before it's ready to merge, mention the PR in the RELEASE-NOTES.md~~

### CODE REVIEW

**Comment by Spaak** [comment]
Thanks @michaelosthege!

I'm wondering what should be the default option. Right now the long `"f(f(f(...)))"` strings are definitely undesirable as defaults, I agree. (Although I think these have always been the default in the LaTeX representation in notebooks, by the way.) If we simplify these to always be compact, I'd say the default should be *with* the params. (Especially in console sessions, simply typing `myvar` it's nice to get `myvar ~ Normal(mu=0.0, sigma=1.0)` without having to explicitly specify a format.)

**Comment by michaelosthege** [comment]
> 
> 
> Thanks @michaelosthege!
> 
> I'm wondering what should be the default option. Right now the long `"f(f(f(...)))"` strings are definitely undesirable as defaults, I agree. (Although I think these have always been the default in the LaTeX representation in notebooks, by the way.) If we simplify these to always be compact, I'd say the default should be _with_ the params. (Especially in console sessions, simply typing `myvar` it's nice to get `myvar ~ Normal(mu=0.0, sigma=1.0)` without having to explicitly specify a format.)

For simple RVs yes, but in most real-world models, these representations will be huge expressions... And they also show up in tracebacks that are already hard to read.

If we adopt a global `rcParams` style output configuration (like ArviZ) the user could change that default.

File: pymc3/distributions/distribution.py
Comment (Spaak) [inline pymc3/distributions/distribution.py:221]: I'm surprised that codecov is complaining that this line is not covered by tests, as far as I can tell it should be?

**Review by Spaak — APPROVED** [review]

**Comment by Spaak** [comment]
Looks good to me! See my one review comment about a line somehow missing tests (or malfunctioning codecov?). Also we're still waiting on Travis before we can merge...

I might file a later PR for your review @michaelosthege which improves (drastically shortens) the display of parameter values, such that maybe that can be the default (again)? For now I agree better to stick with this version.

**Comment by MarcoGorelli** [comment]
> we're still waiting on Travis before we can merge...

Tests are now run via GitHubActions, [Travis can be ignored](https://github.com/pymc-devs/pymc3/pull/4262#issuecomment-734479089)

I'll just have a quick look regarding the non-covered line

EDIT: just noticed there's a test failure on master, I'll take a look at that, no need to wait for me on this one

**Comment by twiecki** [comment]
Thanks @michaelosthege and @Spaak!

**Pull request merged**
